File: tests/indexing/parsing/lang/fixtures/kotlin/Lambdas.kt
package pipeline

class Processor {
    private var onComplete: (() -> Unit)? = null

    fun transform(values: List<Int>): List<Int> {
        val doubler = fun(x: Int): Int { return x * 2 }
        return values
            .filter { it > 0 }
            .map(fun(v: Int): Int {
                val scaled = doubler(v)
                return scaled + 1
            })
    }

    fun configure() {
        this.onComplete = fun() {
            println("done")
        }
    }

    fun summarize(values: List<Int>): Int {
        val total = values.fold(0) { acc, v ->
            acc + v
        }
        return total
    }
}

File: tests/indexing/parsing/lang/fixtures/cpp/Lambdas.cpp
struct Handler {
    std::function<void()> on_done = []() {
        done = true;
        pending -= 1;
    };
};

void configure(std::vector<int> &values) {
    auto doubler = [](int x) {
        int scaled = x * 2;
        return scaled;
    };

    std::function<int(int)> shift = [](int v) {
        int base = v + 10;
        int adjusted = base - 3;
        return adjusted;
    };

    std::sort(values.begin(), values.end(), [](int a, int b) {
        return a > b;
    });

    doubler = [](int x) {
        int bumped = x + 1;
        return bumped * bumped;
    };
}
